PythonVisualizations/Heap.py

```diff
@@ -0,0 +1,518 @@
+        # move value to output box
+        # self.highlightCodeTags('print', callEnviron)
+        toPositions = (outputBoxCoords[0] + padding / 2 + (peekIndex + 1 / 2) * spacing,
+                        midOutputBox)
+        self.moveItemsTo(valueList, (toPositions,), sleepTime=.02)
+        # make the value 25% smaller
+        newSize = (self.VALUE_FONT[0], int(self.VALUE_FONT[1] * .75))
+        self.canvas.itemconfig(valueOutput, font=newSize)
+
+        # add label to output box
+        labelX = self.HEAP_X0 * 2 + boxSize * spacing + padding * 5
+        labelY = canvasDimensions[1] - padding * 2
+
+        boxLabel = self.canvas.create_text(
+            labelX, labelY, text="root", font=self.VARIABLE_FONT)
+        callEnviron.add(boxLabel)
+        self.highlightCodeTags([], callEnviron)
+        # finish animation
+        self.cleanUp(callEnviron)
+
+
+    def removeMax(self):
+        self.startAnimations()
+        callEnviron = self.createCallEnvironment()
+
+        # remove first element from list
+        n = self.list[0]
+        self.list[0] = None
+        self.size -= 1
+        # if the array is now empty, delete the NoneType place holder from the array
+        if self.size < 1: del self.list[:]
+        # Slide value rectangle up and off screen
+        items = (n.display_shape, n.display_val)
+        toPositions = (100, -100, 100, -50)
+        callEnviron |= set(items)
+        self.moveItemsBy(items, delta=(0, -max(400, self.canvas.coords(n.display_shape)[3])), steps=self.CELL_HEIGHT,
+                        sleepTime=.05)
+
+        if self.size < 1:
+            self.setMessage("Heap requirement satisfied")
+
+        # move bottom cell to top, and heapify
+        else:
+            self.swapRoot()
+            self.heapify()
+
+        # Finish animation
+        self.cleanUp(callEnviron)
+
+
+    def isHeap(self):
+        cur = 0
+        size = len(self.list)
+
+        while cur < size // 2:
+            leftChild = 2 * cur + 1
+            rightChild = leftChild + 1
+
+            top = self.list[cur]
+            if self.list[leftChild] > top or (rightChild < size and self.list[rightChild] > top):
+                return False  # return false if a child is larger than a parent
+            else:
+                cur += 1
+
+        return True
+
+
+    def heapify(self):
+        # trickle down until heap is in order
+        while not self.isHeap():
+            self.trickleDown()
+
+
+    def swapRoot(self):
+        self.startAnimations()
+        callEnviron = self.createCallEnvironment()
+        n = self.list[-1]
+        items = (n.display_shape, n.display_val)
+
+        # move the last cell to the front of the heap
+        cellCoords = self.cellCoords2(0)
+        cellCenter = self.cellCenter2(0)
+        toPositions = (cellCoords, cellCenter)
+        # self.moveItemsOffCanvas(items, N, sleepTime=0.02) ## alternative method to moveItemsBy
+        self.moveItemsTo(items, toPositions)

```

Comment:
Let's use the `moveItemsOnCurve()` method to move the last item to the root.  Using the linear movement means the item slides along the column of all the other items in the heap array.  Some of them can be higher in the stacking order and obscure the item being moved.  Also items with the same colored backgrounds can look like they merged.  With the curved trajectory, the moving item should be clearly distinguished from the ones not moving.